bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -574,6 +594,61 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		/// Sends a message to the Gateway contract to register a new
+		/// token that represents `asset`.
+		///
+		/// - `origin`: Must be `MultiLocation` from sibling parachain
+		#[pallet::call_index(10)]
+		#[pallet::weight(T::WeightInfo::register_token())]
+		pub fn register_token(
+			origin: OriginFor<T>,
+			asset: Box<VersionedLocation>,
+			metadata: AssetRegistrarMetadata,
+		) -> DispatchResult {
+			let origin_location: Location = T::SiblingOrigin::ensure_origin(origin)?;
+
+			let (para_id, agent_id) = ensure_sibling::<T>(&origin_location)?;
+
+			let asset: Location =
+				(*asset).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
+
+			let pays_fee = PaysFee::<T>::Yes(sibling_sovereign_account::<T>(para_id));
+
+			Self::do_register_token(para_id, agent_id, asset, metadata, pays_fee)?;
+
+			Ok(())
+		}
+
+		/// Sends a message to the Gateway contract to register a new
+		/// token that represents `asset`.
+		///
+		/// - `origin`: Must be `MultiLocation`
+		#[pallet::call_index(11)]
+		#[pallet::weight(T::WeightInfo::force_register_token())]
+		pub fn force_register_token(
```

Comment:
If we make token registration permissionless, we won't need this force_ extrinsic.